pennylane/tape/qnode.py

```diff
@@ -140,13 +140,16 @@ def __init__(self, func, device, interface="autograd", diff_method="best", **dif
             )
 
         self.func = func
-        self.device = device
+        self.original_device = device
         self.qtape = None
         self.qfunc_output = None
+        self.diff_method = diff_method
 
-        self._tape, self.interface, self.diff_method = self.get_tape(device, interface, diff_method)
+        self._tape, self.interface, diff_method, self.device = self.get_tape(
+            device, interface, diff_method
+        )

```

Comment:
Since it is now possible for the type of tape to be able to change the device :)